File: 2-AddTwoNumbers/2-AddTwoNumbers.php
// Last updated: 8/6/2025, 11:45:36 PM
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution {

    /**
     * @param ListNode $l1
     * @param ListNode $l2
     * @return ListNode
     */
    function addTwoNumbers($l1, $l2) {
        $flag = false;
        $res = 0;

        $node = new ListNode();
        $dummy = new ListNode(0, $node);
        while ($l1 != null || $l2 != null) {
            $temp = 0;
            if ($l1 != null) {
                $temp += $l1->val;
                $l1 = $l1->next;
            }
            if ($l2 != null) {
                $temp += $l2->val;
                $l2 = $l2->next;
            }
            $node->val = $flag ? ($temp + 1)%10 : $temp%10;
            $flag = $flag ? ($temp+1>=10) : $temp >= 10;
            if ($l1 != null || $l2 != null) {
                $node->next = new ListNode();
                $node = $node->next;
            }
        }
        if ($flag) {
            $node->next = new ListNode(1);
        }
        return $dummy->next;
    }
}

File: 2-AddTwoNumbers/2-AddTwoNumbers.java
// Last updated: 8/7/2025, 4:54:30 PM
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        //merge 2 sorted findMedianSortedArrays

        int i = 0, j=0, k=0;
        int[] sorted = new int[nums1.length + nums2.length];
        while (i < nums1.length || j < nums2.length) {
            if (i < nums1.length && j < nums2.length) {
                if (nums1[i] <= nums2[j]) {
                    sorted[k++] = nums1[i++];
                } else {
                    sorted[k++] = nums2[j++];
                }
            } else if (i == nums1.length) {
                sorted[k++] = nums2[j++];
            } else if (j == nums2.length) {
                sorted[k++] = nums1[i++];
            }
        }
        if (sorted.length % 2 == 1) {
            return (double) sorted[(sorted.length)/2];
        }

        return (double) ( sorted[(sorted.length)/2 - 1] + sorted[(sorted.length)/2])/ 2;
    }
}